api/samples/memval_simple.c

```diff
@@ -219,8 +219,7 @@ instrument_mem(void *drcontext, instrlist_t *ilist, instr_t *where, opnd_t ref)
     size = (ushort)drutil_opnd_mem_size_in_bytes(ref, where);
     drx_buf_insert_buf_store(drcontext, trace_buffer, ilist, where, reg_ptr, reg_tmp,
                              OPND_CREATE_INT16(size), OPSZ_2, offsetof(mem_ref_t, size));
-    drx_buf_insert_update_buf_ptr(drcontext, trace_buffer, ilist, where, reg_ptr,
-                                  DR_REG_NULL, sizeof(mem_ref_t));
+    /* postpone updating trace_buffer ptr to post-write, in case the write segfaults */
```

Comment:
But instrument_mem() is called for loads too, and this change means the buffer pointer is never advanced for loads?

Reply:
This[1] loop calls `instrument_mem` only for dests, so seems like it's only for stores? Also this comment[2] documents that `memval_simple` dumps only write addresses.

[1]: https://github.com/DynamoRIO/dynamorio/blob/cf9d6a95262015581a5184e75ce599cc66ac4df4/api/samples/memval_simple.c#L356
[2]: https://github.com/DynamoRIO/dynamorio/blob/cf9d6a95262015581a5184e75ce599cc66ac4df4/api/samples/memval_simple.c#L36

Reply:
I see.  Maybe this should be called instrument_pre_write() rather than instrument_mem(): that would make it clearer.

Reply:
Done. Renamed.